src/main/scala/io/iohk/ethereum/network/ConnectedPeers.scala

```diff
@@ -65,11 +74,49 @@ case class ConnectedPeers(
 
     (
       peersId,
-      ConnectedPeers(incomingPendingPeers -- peersId, outgoingPendingPeers -- peersId, handshakedPeers -- peersId)
+      ConnectedPeers(
+        incomingPendingPeers -- peersId,
+        outgoingPendingPeers -- peersId,
+        handshakedPeers -- peersId,
+        pruningPeers -- peersId,
+        lastPruneTimestamp = lastPruneTimestamp
+      )
     )
   }
+
+  def prunePeers(
+      incoming: Boolean,
+      minAge: FiniteDuration,
+      numPeers: Int,
+      currentTimeMillis: Long = System.currentTimeMillis
+  ): (Seq[Peer], ConnectedPeers) = {
+    val ageThreshold = currentTimeMillis - minAge.toMillis
+    if (lastPruneTimestamp > ageThreshold || numPeers == 0) {
+      // Protect against hostile takeovers by limiting the frequency of pruning.
+      (Seq.empty, this)
+    } else {
+      val candidates = handshakedPeers.collect {
+        case (_, p)
+            if (p.incomingConnection || !incoming)
+              && p.createTimeMillis <= ageThreshold
+              && !pruningPeers.contains(p.id) =>
+          p
+      }.toSeq
+
+      val toPrune = Random.shuffle(candidates).take(numPeers)
```

Comment:
I have a mixed feeling about disconnecting randomly form peers, especially those which are connected to the node for a pretty long time as those usually provide most data and are most stable
Imo there are two forces at play:

-From network point of view this makes sense, as it makes space for new joiners.
-From node point of view it makes no sense, as node will randomly disconnect from peers to which it was connected pretty long time (so most probably most stable ones)

Maybe we should couple this strategy with some kinda tracking when was the last communication to given peer, and select only peers which do not provide us any data or do not communicate with us in any way?

Reply:
Definitely that would be a great improvement, I just didn't want to touch even more components in this PR and introduce even more communication with `PeerActor` instances where I assume this information would be stored. 

The protection from rapidly throwing out old peers is that it will not do pruning more often than once per 30 minutes, so it would take some time until all the old guard gets randomly cleared out.

Reply:
Do you want this added to this PR, or a separate one where we can hash out the details?

Reply:
Created a ticket ETCM-463 to maintain statistics.

Reply:
See https://github.com/input-output-hk/mantis/pull/849 for a mechanism we could use for prioritisation.